src/bottom-navigation-bar/lib/android/listeners/on-tab-selected.listener.ts

```diff
@@ -0,0 +1,31 @@
+import { BottomNavigationBar } from '../bottom-navigation-bar';
+
+// Classes shortcuts
+const { BottomNavigationView } = com.google.android.material.bottomnavigation;
+
+// Interface shortcuts
+declare type OnNavigationItemSelectedListenerType = com.google.android.material.bottomnavigation.BottomNavigationView.OnNavigationItemSelectedListener;
+
+@Interfaces([BottomNavigationView.OnNavigationItemSelectedListener])
+export class OnTabSelectedListener extends java.lang.Object
+    implements OnNavigationItemSelectedListenerType {
+    private _owner: BottomNavigationBar;
+
+    static initWithOwner(owner: WeakRef<BottomNavigationBar>) {
+        const listener = new OnTabSelectedListener();
+        listener._owner = owner.get();
+        return listener;
+    }
+
+    public onNavigationItemSelected(item: android.view.MenuItem): boolean {
+        const bottomNavigationTab = this._owner.items[item.getItemId()];
+
+        if (bottomNavigationTab.isSelectable) {
+            this._owner._emitTabSelected(item.getItemId());
+        } else {
+            this._owner._emitTabPressed(item.getItemId());
+        }
+
+        return bottomNavigationTab.isSelectable;
+    }
+}
```

Comment:
Yes, I agree with you, I already made some changes regarding that and it works like a charm, but I want to do some tests first.

Regarding of having the code in the same file is not a good practice, the code is not easy to read or understand, there would be consequences, it was already tested with webpack and even in release mode with snapshot, remember that at the end all the code is merged in one by webpack, so we don't need to do that and have a more organized code.

Reply:
@henrychavez I can agree it is not the best practice. But i will keep that way for now. Especially because this is the way the {N} team does it. And this plugin need to conform to their way for now. Sorry ...
Might think about changing that in the future. But again want to do it for all packages at the same time